spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/outbound/PubSubMessageHandler.java

```diff
@@ -89,16 +88,18 @@ protected void handleMessageInternal(Message<?> message) throws Exception {
 
 		ByteString pubsubPayload;
 
-		if (payload instanceof byte[]) {
+		if (this.messageConverter != null) {
+			pubsubPayload = ByteString.copyFrom((byte[]) this.messageConverter.fromMessage(message,
+					byte[].class));
+		}
+		else if (payload instanceof byte[]) {
 			pubsubPayload = ByteString.copyFrom((byte[]) payload);
 		}
 		else if (payload instanceof ByteString) {
 			pubsubPayload = (ByteString) payload;
 		}
 		else {
-			pubsubPayload =	ByteString.copyFrom(
-					(String) this.messageConverter.fromMessage(message, String.class),
-					Charset.defaultCharset());
+			pubsubPayload = ByteString.copyFrom(message.toString().getBytes());
```

Comment:
Instead of this not so friendly operation I suggest to have a `GenericMessageConverter` as a default converter and don't allow to overrride it to `null`.

The `GenericMessageConverter` is base on the Spring's `ConversionService` and provide a lot of options to convert from one type to another: https://docs.spring.io/spring/docs/5.0.6.RELEASE/spring-framework-reference/core.html#core-convert-ConversionService-API

Reply:
Should we do the same in the `PubSubInboundChannelAdapter`?

Reply:
No, because we don't know the target type preferences. Just clear the situation in the Docs (and JavaDocs) that without any converter we are going to produce a message with just `byte[]` payload.

Reply:
The weird thing is that we also have the `PubSubMessageConverter` and that is only used in the `PubSubTemplate`. Shouldn't we try to reuse it somehow in the channel adapters?

Reply:
I think it doesn't matter here on the `PubSubMessageHandler` since both `PubSubMessageConverter` and Spring's `MessageConverter` are going to do the same functionality - extract body from the `Message`. I would still vote for the `MessageConverter` just because we deal here with the Spring `Message<?>` meanwhile `PubSubTemplate` doesn't.

Reply:
Could `GenericMessageConverter` also fail to convert `Message` to `byte[]`? What should we do then?

Reply:
In this case we throw an exception to the caller notifying him/her that the payload is not compatible for sending to Pub/Sub. Fair? We may mention such a limitation in the Docs. 
Otherwise all this `toString().getBytes()`  does not make sense.

Reply:
If we make the converter required, then we never fall into the other branches of the `if` that allow for `byte[]` and `ByteString`. How would you handle that?

Reply:
You need to swap that `if...else` logic then. No reason to convert to `byte[]` if it is already `byte[]`. Isn't it ?

Reply:
I'm not sure. Maybe someone would want to because `byte[]` can be anything. We would also then need to explain that the converter is ignored if the payload is `byte[]` or `ByteString`. i'm fine with that, if you are.

Reply:
Well, we have some convenient built-in functionality, anything else, fancy enough, can be done in advance before sending Spring message to this component. Remember: we deal here with Spring Integration, so any complexity an be done with other endpoints? Also would be great to let to send a `PubSubMessage` here per se. I mean as `payload` of the Spring message. Some custom `transfomer` upfornt can buid that for us.

Reply:
Agreed, fancy transformations can be done in the SI flow before it reaches the adapter. I'll update accordingly. Thanks!